<?xml version="1.0"?>
<launch>

	<!-- arguments -->
	<arg name="pddl_file_path" 	default="$(find rosplan_agent_interface)/test/leak_response/"/>
	<arg name="domain_path"		default="$(arg pddl_file_path)domains/leak_failure.pddl" />
	<arg name="problem_path"	default="$(arg pddl_file_path)problems/leak_failure_problem.pddl" />
	<arg name="planner_command"	default="$(find rosplan_planning_system)/common/bin/lpg-td -o DOMAIN -f PROBLEM -quality -out $(arg pddl_file_path)/plans/plan.pddl" /> 

	<!-- Parsing -->
	<node name="rosplan_parsing_interface" pkg="rosplan_planning_system" type="pddl_simple_plan_parser" respawn="false" output="screen">
		<param name="knowledge_base"	value="rosplan_knowledge_base"/>
		<param name="planner_topic"		value="/rosplan_planner_interface/planner_output"/>
		<param name="plan_topic"		value="complete_plan"/>
	</node>

	<!-- Configurator -->
	<include file="$(find rosplan_agent_interface)/launch/configurator_interface.launch">
		<arg name="node_name"			value="rosplan_configurator_interface"/>
		<arg name="goal_request_topic"	value="goals"/>
		<arg name="problem_topic"		value="problems"/>
		<arg name="planner_command"		value="$(arg planner_command)"/>
		<arg name="pddl_file_path"		value="$(arg pddl_file_path)"/>
	</include>

	<!-- KB -->
	<node name="rosplan_knowledge_base" pkg="rosplan_knowledge_base" type="knowledgeBase" respawn="false" output="screen">
		<param name="domain_path"	value="$(arg domain_path)"/>
		<param name="problem_path"	value="$(arg problem_path)"/>
		<!-- conditional planning flags -->
		<param name="use_unknowns"	value="false"/>
	</node>

	<!-- Executive -->
	<include file="$(find rosplan_agent_interface)/launch/executive_interface.launch">
		<arg name="node_name"			value="rosplan_executive_interface"/>
		<arg name="configurator_topic"	value="/rosplan_configurator_interface/output"/>
		<arg name="dispatch_topic"		value="output"/>
	</include>

	<!-- Dispatch -->
	<node name="rosplan_plan_dispatcher" pkg="rosplan_planning_system" type="pddl_simple_plan_dispatcher" respawn="false" output="screen">
		<param name="knowledge_base"	value="rosplan_knowledge_base"/>
		<param name="plan_topic"		value="/rosplan_executive_interface/output"/>
		<param name="action_dispatch_topic"	value="action_dispatch"/>
		<param name="action_feedback_topic"	value="action_feedback"/>
	</node>


</launch>
